File: Framework/HackerNews.Framework/Classes/Story.cs
using HackerNews.Framework.Interfaces;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace HackerNews.Framework.Classes
{
    public class Story : IStory
    {
        [JsonProperty("by")]
        public string By { get; set; }
        [JsonProperty("descendants")]
        public string Descendants { get; set; }
        [JsonProperty("id")]
        public int Id { get; set; }
        [JsonProperty("kids")]
        public List<int> Kids { get; set; }
        [JsonProperty("score")]
        public int Score { get; set; }
        [JsonProperty("text")]
        public string Text { get; set; }
        [JsonProperty("time")]
        public int Time { get; set; }
        [JsonProperty("title")]
        public string Title { get; set; }
        [JsonProperty("type")]
        public string Type { get; set; }
        [JsonProperty("url")]
        public string Url { get; set; }

    }
}

File: ClientApp/src/app/classes/story.ts
export class Story implements IStory {    
    by: string;
    descendants: number;
    id: number;
    kids: Array<number>;
    score: number;
    text: string;
    time: Date;
    title: string;
    type: string;
    url: string; 

    constructor(data: any = null) {
        if (data) {
            let time = new Date(0);
            time.setUTCSeconds(data.time);

            this.by = data.by;
            this.descendants = data.descendants;
            this.kids = data.kids;
            this.score = data.score;
            this.text = data.text;
            this.time = time;
            this.title = data.title;
            this.type = data.type;
            this.url = data.url;
        }
    }
}